scripts/check-users-in-space-developer-role.ps1

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env pwsh
+
+param([String] $space, [String] $slackwebhook)
+
+$data = @()
+
+$bearer=$(cf oauth-token)
+
+$headers = @{
+    Authorization = "$bearer"
+    Accept = "application/json"
+  }
+
+$get_space_details=Invoke-RestMethod "https://api.london.cloud.service.gov.uk/v3/spaces?names=$space"  -H $headers
+$space_guid=$get_space_details.resources[0].guid
+
+$get_space_details=Invoke-RestMethod "https://api.london.cloud.service.gov.uk/v3/roles?space_guids=$space_guid&types=space_developer"  -H $headers
+
+$get_developer_ids=$get_space_details.resources.relationships.user.data.guid
+
+  Foreach ($id in $get_developer_ids){
+    $num += 1
+    $user=Invoke-RestMethod "https://api.london.cloud.service.gov.uk/v3/users?guids=$id" -H $headers
+    Write-Output $user.resources.username
+    $data +=  $num.ToString() + (". ") + $user.resources.username + "`n"
+
+  }
+
+foreach ($item in $data) {
+    write-host $item
+}
+
+$uriSlack = $slackwebhook
+$body = ConvertTo-Json @{
+    text = ":warning: The following users have the SpaceDeveloper role in PaaS space $space :- `n" + $data
+    type = "mrkdwn"
+}
+
+write-host $body
+
+try {
+    Invoke-RestMethod -uri $uriSlack -Method Post -body $body -ContentType 'application/json' | Out-Null
+} catch {
+    Write-Error (Get-Date) ": Update to Slack went wrong..."
```

Comment:
Will we see anything other than `2021-03-02: Update to Slack went wrong...` if there's an error? Not sure if PowerShell has a concept of standard error but the `Invoke-RestMethod` above appears to bin the output.

Reply:
```powershell
    $ErrorMessage = $_.Exception.Message
    $FailedItem = $_.Exception.ItemName
```

Maybe that....

Reply:
yes there is.... i will include this....

Reply:
you now get something like this

`Write-Error: 03/04/2021 10:25:27
Write-Error: System.Management.Automation.ParameterBindingException: A parameter cannot be found that matches parameter name 'ur1i'.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception) in /PowerShell/src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs:line 1655
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame) in /PowerShell/src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs:line 504
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) in /PowerShell/src/System.Management.Automation/engine/interpreter/ControlFlowInstructions.cs:line 392
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) in /PowerShell/src/System.Management.Automation/engine/interpreter/ControlFlowInstructions.cs:line 358`